lib/multicell_location/services/nrf_cloud_integration.c

```diff
@@ -60,6 +69,48 @@ const char *location_service_get_certificate_nrf_cloud(void)
 	return tls_certificate;
 }
 
+#if defined(CONFIG_NRF_CLOUD_MQTT)
+static void location_service_location_ready_cb(const struct nrf_cloud_cell_pos_result *const result)
+{
+	if (result != NULL) {
+		nrf_cloud_location.latitude = result->lat;
+		nrf_cloud_location.longitude = result->lon;
+		nrf_cloud_location.accuracy = (double)result->unc;
+	}
+	k_sem_give(&location_ready);
+}
+
+int location_service_get_cell_location_nrf_cloud(
+	const struct lte_lc_cells_info *cell_data,
+	char * const rcv_buf,
+	const size_t rcv_buf_len,
+	struct multicell_location *const location)
+{
+	ARG_UNUSED(rcv_buf);
+	ARG_UNUSED(rcv_buf_len);
+
+	int err;
+
+	err = nrf_cloud_cell_pos_request(cell_data, true, location_service_location_ready_cb);
+	if (err) {
+		LOG_ERR("Failed to request cell pos data, error: %d", err);
+		return err;
+	}
+
+	LOG_INF("Cellular positioning request sent");
+
+	if (k_sem_take(&location_ready, K_SECONDS(20)) == -EAGAIN) {
+		LOG_ERR("Cellular positioning data request timed out");
+		return -ETIMEDOUT;
+	}
+
+	location->latitude = nrf_cloud_location.latitude;
+	location->longitude = nrf_cloud_location.longitude;
+	location->accuracy = nrf_cloud_location.accuracy;
```

Comment:
Would this work as it's a simple structure without pointers:
`*location = nrf_cloud_location;`

Reply:
Yes, thanks.